### PR TITLE
ENH run_in_pyodide type annotations, defaults, posonly, and kwonly args

### PR DESCRIPTION
**Pull request by hoodmane**
It still doesn't support *args or **kwargs arguments though.

- [x] Add tests
- [ ] Update changelog

### CODE REVIEW

**Comment by ryanking13** [comment]
Thanks for keep improving `run_in_pyodide`! For the part about modifying the AST, it's hard to understand if it's correct just by looking at the code... so I'll review it again when the tests are added.

**Comment by hoodmane** [comment]
Okay I added tests. Of course it didn't work very well before I added the tests.

I reorganized things a bit to reduce the reliance on class attributes when it can be avoided and make things a bit easier to read. Unfortunately this made the diff a bit big.

File: tests/test_decorator.py [hunk 1]
```diff
@@ -296,3 +298,57 @@ def helper(selenium, meth):
...
+    wrapped(selenium, 5, Callable)
+    assert wrapped.__annotations__ == fn_with_typehints.__annotations__
+
+
+def test_posonly(selenium):
+    @run_in_pyodide
+    def fn_with_posonly(selenium, a, /, b):
+        return None
+
+    fn_with_posonly(selenium, 5, 7)
```
Comment (ryanking13) [inline tests/test_decorator.py:322]: Could you also check the values allocated a and b are correct?

**Review by ryanking13 — APPROVED** [review]
Thanks! I should admit that I don't fully understand the AST hacks. Anyway, the test passes so I am okay to merge this.

**Pull request merged**